baseplate/ratelimit/backends/base.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import time
+
+
+class RateLimitBackend(object):
+    """An interface for rate limit backends to implement.
+
+    :param str key: The name of the rate limit bucket to consume from.
+    :param int amount: The amount to consume from the rate limit bucket.
+    :param int allowance: The maximum allowance for the rate limit bucket.
+    :param int interval: The interval to reset the allowance.
+
+    """
+    def consume(self, key, amount, max, bucket_size):
+        raise NotImplementedError
+
+    @staticmethod
+    def _get_current_bucket(bucket_size):
```

Comment:
can this just be a top-level function? even though it's not being used that way, seeing it on the class made me worry about implementation inheritance

Reply:
I can't find my feedback about this, I made this suggestion so each subclass didn't have to import both this class and the top level function.

Reply:
If you prefer it to be `get_current_bucket` that's fine too, I don't feel strongly about it :)